src/routing_table/mod.rs

```diff
@@ -385,6 +385,47 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
+    /// Are we among the `count` closest nodes to `name`?
+    pub fn is_closest(&self, name: &T, count: usize) -> bool {
+        self.closest_names(name, count).is_some()
+    }
+
+    /// Returns the `count` closest entries to `name` in the routing table, including our own name,
+    /// sorted by ascending distance to `name`. If we are not close, returns `None`.
+    pub fn closest_names(&self, name: &T, count: usize) -> Option<Vec<&T>> {
+        let result = self.groups
```

Comment:
This could be reduced a bit to:
```rust
        let result = self.iter()
            .chain(iter::once(&self.our_name))
            .sorted_by(|name0, name1| name.cmp_distance(name0, name1))
            .into_iter()
            .take(count)
            .collect_vec();
```